pytensor/link/pytorch/dispatch/elemwise.py

```diff
@@ -34,3 +35,46 @@ def dimshuffle(x):
         return res
 
     return dimshuffle
+
+
+@pytorch_funcify.register(Softmax)
+def pytorch_funcify_Softmax(op, **kwargs):
+    axis = op.axis
+
+    def softmax(x):
+        if not torch.is_floating_point(x):
+            x = x.to(torch.float32)
```

Comment:
Should probably convert to the output type advertised by PyTensor. The second argument of the funcify function, is `node` (now hidden inside **kwargs. If you retrieve `node`, you can check the dtype of the output via `node.outputs[0].dtype`. This is probably what you should convert to, not necessarily float32
```suggestion
            x = x.to(torch.float32)
```

Reply:
Actually there's a bug in PyTensor Softmax, it also fails if you try to execute with dtype. Can you open a separate issue for that in the repo here? For now it's enough for the torch dispatch function to raise a `NotImplementedError` if the output dtype is an integer, and not try to handle it inside the returned `softmax` function.

Same for the other Softmax related functions

Reply:
Opened an issue: https://github.com/pymc-devs/pytensor/issues/857